File: backend/app/schemas/imagenes.py
from pydantic import BaseModel


class ImagenListItem(BaseModel):
    serial: str
    nombred: str | None = None
    dirdes1: str | None = None
    ciudad1: str | None = None
    f_emi: str | None = None
    cod_men: str | None = None
    ret_esc: str | None = None
    motivo: str | None = None


class ImagenGuia(BaseModel):
    serial: str
    encontrado: bool
    image_url: str
    no_entidad: str | None = None
    servicio: str | None = None
    nombred: str | None = None
    dirdes1: str | None = None
    ciudad1: str | None = None
    cod_sec: str | None = None
    retorno: str | None = None
    ret_esc: str | None = None
    orden: str | None = None
    planilla: str | None = None
    f_emi: str | None = None
    f_lleva: str | None = None
    cod_men: str | None = None
    dir_num: str | None = None
    comentario: str | None = None

File: frontend/src/api/imagenes.ts
import api from "./client";

export interface ImagenGuia {
  serial: string;
  encontrado: boolean;
  image_url: string;
  no_entidad: string | null;
  servicio: string | null;
  nombred: string | null;
  dirdes1: string | null;
  ciudad1: string | null;
  cod_sec: string | null;
  retorno: string | null;
  ret_esc: string | null;
  orden: string | null;
  planilla: string | null;
  f_emi: string | null;
  f_lleva: string | null;
  cod_men: string | null;
  dir_num: string | null;
  comentario: string | null;
}

export interface ImagenListItem {
  serial: string;
  nombred: string | null;
  dirdes1: string | null;
  ciudad1: string | null;
  f_emi: string | null;
  cod_men: string | null;
}

export type ImagenesModo = "serial" | "nombre" | "direccion";

export const imagenesApi = {
  obtenerImagen: (serial: string) =>
    api.get<ImagenGuia>(`/imagenes/${encodeURIComponent(serial)}`),
  obtenerFoto: (serial: string) =>
    api.get(`/imagenes/${encodeURIComponent(serial)}/foto`, { responseType: "blob" }),
  buscarLista: (q: string, modo: "nombre" | "direccion") =>
    api.get<ImagenListItem[]>("/imagenes", { params: { q, modo } }),
};
